### ISSUE TITLE
Provide documentation/syntax for messages

### ISSUE THREAD

**Issue by vmassol** [vector-im/riot-web#4295]
Hi, it would be great if there could be some in-product help about the available syntax that can be used for messages. We've gathered the following so far:

* To type a message with multiple lines, use shift+enter to create a new line inside of your message, use regular enter key in order to send your message
* To send a message with some code on one line, use \```triple-backtick``` around your code (like in markdown)
* To paste a whole bunch of code in a message: triple-backtick, shift+enter, paste, shift+enter, triple-backtick and finally enter to send the message
* To make some __italic__ content in a message, put that content inside of \*stars*
* To make some **bold** content in a message, put that content inside of \*\*double-stars**
* Emoticons are not included in matrix by default, but emoji work.  Mac users can make emoji by typing  alt+command+space  and then selecting from an emoji pallette. Everyone can copy/paste it if necessary. You can also use colon and parenthecy :)
    * If you want to make emoji, you need to go to http://unicode.org/emoji/charts/full-emoji-list.html and copy the emoji you want from the *first* column but remember that in matrix it will appear the way it shows in the *fifth* column (because riot.im app uses the *One* emoji rendering library).
    * To personalize your nickname, use /nick yourname and paste the emoji characters in your name. Note that putting emoji at the *beginning* of your name will make it difficult for people to use tab-complete to message you.

Are there any more? :)

I couldn't find any documentation page on the web with this information either. But IMO it would be the nicest if it could be included in the product.

WDYT?

**Comment by t3chguy** [comment]
All information is here: http://commonmark.org/help/

**Comment by t3chguy** [comment]
Also emoticons are supported through the "New Composer"/RTE
available in Labs in Settings

**Comment by t3chguy** [comment]
but also to go with the latter, RTE will make everything more *graphical* so would render most of the help needed less needed...hopefully...

**Comment by vmassol** [comment]
> All information is here: http://commonmark.org/help/

:) Not exactly true since headings don't work for example :) Are there other commonmark syntax elements not supported?

**Comment by t3chguy** [comment]
Headings **DO** work ;)
![image](https://user-images.githubusercontent.com/2403652/27084517-1c67653a-5044-11e7-991a-bd056a023384.png)

**Comment by vmassol** [comment]
> but also to go with the latter, RTE will make everything more graphical so would render most of the help needed less needed...hopefully...

Cool stuff, eager to try it out. Our instance is probably too old for this RTE thing (a month or so old). We need to upgrade!

**Comment by t3chguy** [comment]
Riot uses a commonmark lib for the md->html so all commonmark stuff *should* work

**Comment by vmassol** [comment]
> Headings DO work ;)

hmm just tried it and it didn't work for me. I typed "# test" and "## test".

**Comment by cjdelisle** [comment]
I think I agree that these things should be something you discover from using the app, not something you need to go to read documentation for.

**Comment by t3chguy** [comment]
@vmassol I think the catch is that they all map to <h3> because <h1>s would have been too intrusively gigantic

**Comment by vmassol** [comment]
ok, got it. Remote images don't work either but @cjdelisle explained why: "remote images don't work because it would be a leak of IP addresses. because people hit the server with the image". 

Thanks for your fast replies ;)

**Comment by t3chguy** [comment]
Yeah, I believe that RTE will become default and completely replace the current "Old Composer" so will solve most of these issues

As to remote images, could parse and try to download them in-browser and re-upload them but that could be broken by CORS etc

**Comment by turt2live** [comment]
Related: #1631

**Comment by pbondoer** [comment]
Related/possible duplicate of #3728

**Comment by lampholder** [comment]
Since #3728 asks just to link to the docs and this is asking for some hand-rolled in-app help, I think we can say that they're solving the same problem but in slightly different ways.

Therefore I reckon this stays open as a feature request releated to #3728 :)

**Comment by uhoreg** [comment]
Should this issue be lower priority now that RTE is default?
